qubesappmenus/__init__.py

```diff
@@ -179,21 +179,31 @@ def get_available_filenames(self, vm):
                 listed.add(filename)
                 yield os.path.join(template_dir, filename)
 
-    def get_available(self, vm):
+    def get_available(self, vm, fields=None):
         '''Get available menu entries for given VM
 
-        Returns a generator of (filename, display_name) tuples'''
+        Returns a generator of lists that contain fields to be outputted'''
         # TODO icon path (#2885)
         for filename in self.get_available_filenames(vm):
+            field_values = {}
             with open(filename) as file:
                 name = None
                 for line in file:
                     if line.startswith('Name=%VMNAME%: '):
                         name = line.partition('Name=%VMNAME%: ')[2].strip()
-                        break
+                    if fields:
+                        for field in fields:
+                            if line.startswith(field + '=') or\
+                                    line.startswith(field + ' ='):
```

Comment:
Space handling around `=` looks wrong, especially since `partition()` call below doesn't do that. Better use `line.split('=', 1)` and strip spaces from result

Reply:
pushed a fix